pkg/controller/awsfederatedaccountaccess/awsfederatedaccountaccess_controller.go

```diff
@@ -309,23 +309,29 @@ func (r *ReconcileAWSFederatedAccountAccess) createIAMRole(awsClient awsclient.C
 	return nil
 }
 
-func (r *ReconcileAWSFederatedAccountAccess) createOrUpdateIAMPolicy(awsClient awsclient.Client, afr awsv1alpha1.AWSFederatedRole, afaa awsv1alpha1.AWSFederatedAccountAccess) error {
+func (r *ReconcileAWSFederatedAccountAccess) createOrUpdateIAMPolicy(awsClient awsclient.Client, afr awsv1alpha1.AWSFederatedRole) error {
 
-	policy, err := r.createIAMPolicy(awsClient, afr, afaa)
+	err := CreateOrUpdateIAMPolicy(awsClient, afr)
+
+	return err
+}
+
+func CreateOrUpdateIAMPolicy(awsClient awsclient.Client, afr awsv1alpha1.AWSFederatedRole) error {
+
+	policy, err := createIAMPolicy(awsClient, afr)
 	if err != nil {
 		if aerr, ok := err.(awserr.Error); ok {
 			if aerr.Code() == "EntityAlreadyExists" {
 
 				// If the Role already exists, delete it and recreate it
 				_, err = awsClient.DeletePolicy(&iam.DeletePolicyInput{PolicyArn: policy.Arn})
-				_, err = r.createIAMPolicy(awsClient, afr, afaa)
+				_, err = createIAMPolicy(awsClient, afr)
```

Comment:
These errors need to be handled properly or passed forward properly to the next func.